src/installer.ts

```diff
@@ -338,10 +370,112 @@ async function resolveVersionFromManifest(
   }
 }
 
+function evaluateNightlyVersions(
+  versions: string[],
+  versionSpec: string
+): string {
+  let version = '';
+  let range: string | null | undefined;
+  const [raw, prerelease] = versionSpec.split('-');
+  const isValidVersion = semver.valid(raw);
+  const rawVersion = isValidVersion ? raw : semver.coerce(raw);
+  if (rawVersion) {
+    if (prerelease !== 'nightly') {
+      range = `${rawVersion}+${prerelease.replace('nightly', 'nightly.')}`;
+    } else {
+      range = semver.validRange(`^${rawVersion}`);
+    }
+  }
+
+  if (range) {
+    versions = versions.sort((a, b) => {
+      if (semver.gt(a, b)) {
+        return 1;
+      }
+      return -1;
+    });
```

Comment:
There's a `compare` function you can pass to `Array.prototype.sort`: https://github.com/npm/node-semver#comparison

```suggestion
    versions = versions.sort(semver.compare));
```